extensions/chathistory.md

```diff
@@ -0,0 +1,106 @@
+
+Both the `limit` and `max_messages` MUST be integers, and `max_messages` MUST be greater than or equal to zero. The client SHOULD NOT request a `limit` with an absolute value greater than the `max_messages` parameter sent in the command. If the `limit` is 0 or no `limit` is provided, the number of messages equal to `max_messages` MUST BE returned.
+
+Both the `start` and `end` parameters support `draft/msgid` and `timestamp`. If the number of lines between the `start` and `end` parameters exceeds the `max_messages`, the server SHOULD return a number of lines equal to the `max_messages` and the appropriate warning as described below. A`limit` or `max_messages` of 0 indicates that no limit or maximum exists.
+
+The `target` parameter specifies a single channel or query from which history SHOULD be retrieved. Wildcards or multiple targets are not supported.
+
+#### Context
+
+The following keywords are used to describe how the server should return messages relative to the `timestamp(s)` or `draft/msgid(s)` given.
+
+`BEFORE` get up to `limit` number of messages before the given `timestamp` or `draft/msgid`. The `limit` MUST BE positive.
+
+`AFTER` get up to `limit` number of messages after the given `timestamp` or `draft/msgid`. The `limit` MUST BE positive.
+
+`LATEST` get the most recent (up to `limit`) messages that have been sent since the given `timestamp` or `draft/msgid`. The `limit` MUST BE positive.
+
+`AROUND` get a number of messages before and after the `timestamp` or `draft/msgid`. The `limit` MUST BE positive.
+
+`BETWEEN` get up to `limit` number of messages between the given `timestamps` or `draft/msgids`. The `limit` MAY positive or negative. If the `limit` is positive, messages will be returned in ascending order starting at the oldest message. If the `limit` is negative, messages will be returned in decending order starting at the newest message. 
+
+#### Format
+`chathistory` uses the following generic format:
+
+    @draft/label=ID CHATHISTORY <target> <context> <start> <end> [<direction>] [<limit>]
+
+The `chathistory` content can requested using timestamps:
+
+    @draft/label=ID CHATHISTORY <target> <context> timestamp=YYYY-MM-DDThh:mm:ss.sssZ [<direction>] [<limit>]
+
+Alternatively, content can be requested using a `draft/msgid`:
+
+    @draft/label=ID CHATHISTORY <target> <context> draft/msgid=<message_id> timestamp=<timestamp>
+
+For `BEFORE`, `AFTER`, `LATEST`, and `AROUND`, a single `timestamp` or `draft/msgid` is required.  For `BETWEEN`, both a `start` and `end` `timestamp` and/or `draft/msgid` is required. The  `limit` is optional for all contexts.
+
+#### Errors and Warnings
+If the server receives a `CHATHISTORY` command with missing parameters, the `NEED_MORE_PARAMS` error code SHOULD be returned.
+
+If the number of lines between the `start` and `end` parameters exceeds the `max_messages`, warn code `MAX_MESSAGES_EXCEEDED` SHOULD be returned. The command SHOULD continue to be processed as described above.
```

Comment:
Ehh, this doesn't feel too appropriate as a warning. This is gonna be a pretty standard occurrence when using this subcommand (and honestly the case of 'more playback exists' is gonna happen in other subcommands too).

Maybe this would make sense as a `MORE_HISTORY_EXISTS` or something better-named instead, and not really either a warning or an error.